src/main/java/com/google/sps/data/User.java

```diff
@@ -6,14 +6,17 @@
 	private final String password;
 	private final String email;
 	private final LocalDateTime lastLogin;
+	private final boolean appearingOnline;
```

Comment:
This name is a bit confusing, I should be able to figure out what the field does from its name. You actually assign it to "isActive" in other places suggesting the original name is not clear enough.

Reply:
To clarify on lastLogin, appearingOnline, their relation and why the name is appearingOnline here is how we are using them. When a user logs in we update their lastLogin, which is a LocalDateTime value converted to a string. appearingOnline is a boolean relating to a common feature in online platforms where users can choose to "appear offline" even when they are logged in. When the user logs in (and chooses to appearOnline) we update their lastLogin time every minute. We then check that their lastLogin time is within a minute (might change to two)  of the current time upon the backend call and use this to measure whether a user is "active" or not. However, if a user sets their appearingOnline status to false (or chooses to appearOffline) then they are treated as an "inactive" user even if they are logged into the application. isActive would be more of confusing name and doesn't correspond to how the field is used because we not checking whether they are active we are checking whether they _want_ people to know when they are active

Reply:
Thanks for the clarification. My point is that I should be able to look at the code and understand what's happening rather than read a comment. Same goes for method and class names. You want the code to be easy to read if someone else has to make changes. 
So, I'd still recommend changing the variable name (something along the lines of offlinePresence/onlinePresence or even isOffline). You could even use a different type instead of boolean to distinguish between the states i.e enum.

Reply:
Maybe we could change lastLogin to lastActive if that would make things clearer